### PR TITLE
Introduce Public Alpha

### PR DESCRIPTION
**Pull request by migueldingli1997**
Existing alpha is now 'system' alpha and introduced new 'public' alpha, where the 'system' alpha is now calculated from the 'public' alpha. Note that `MsgSetNextAlpha` now updates the 'public' alpha. The 'system' alpha, rather than being supplied by this message, is now calculated based on the 'public' alpha. The starting 'public' alpha is 0.5, stored in `StartingPublicAlpha`.

### CODE REVIEW

File: x/bonds/handler.go [hunk 1]
```diff
@@ -333,32 +336,64 @@ func handleMsgSetNextAlpha(ctx sdk.Context, keeper keeper.Keeper, msg types.MsgS
...
+		newSystemAlpha = sdk.OneDec().Sub(temp1.Mul(temp2))
+	} else {
+		// (1 - scaled_delta_public_alpha) * (previous_alpha)
+		temp1 := sdk.OneDec().Sub(scaledDeltaPublicAlpha)
+		temp2 := prevSystemAlpha
+		newSystemAlpha = temp1.Mul(temp2)
+	}
+
+	// Check 1 (newSystemAlpha != systemAlpha)
+	if newSystemAlpha.Equal(paramsMap["systemAlpha"]) {
```
Comment (stefaniatadama) [inline x/bonds/handler.go:372]: Could have used `prevSystemAlpha`?

Comment (stefaniatadama) [inline x/bonds/handler.go:372]: Could we have used `prevSystemAlpha` here?

File: x/bonds/handler.go [hunk 2]
```diff
@@ -333,32 +336,64 @@ func handleMsgSetNextAlpha(ctx sdk.Context, keeper keeper.Keeper, msg types.MsgS
...
+	}
+	// Check 2 (I > C * systemAlpha)
+	if paramsMap["I0"].LTE(newSystemAlpha.MulInt(C)) {
 		return nil, sdkerrors.Wrap(types.ErrInvalidAlpha,
 			"cannot change alpha to that value due to violated restriction [1]")
 	}
-	// Check 3 (R / C > newAlpha - alpha)
-	if R.QuoInt(C).LTE(newAlpha.Sub(paramsMap["alpha"])) {
+	// Check 3 (R / C > newSystemAlpha - systemAlpha)
+	if R.QuoInt(C).LTE(newSystemAlpha.Sub(paramsMap["systemAlpha"])) {
```
Comment (stefaniatadama) [inline x/bonds/handler.go:382]: Could we have used `prevSystemAlpha` here too?

**Review by stefaniatadama — APPROVED** [review]

File: x/bonds/handler.go (hunk 1 above)
Comment (migueldingli1997) [reply x/bonds/handler.go]: Yep, thanks!

File: x/bonds/handler.go (hunk 2 above)
Comment (migueldingli1997) [reply x/bonds/handler.go]: Yep, thanks!

Comment (migueldingli1997) [reply x/bonds/handler.go]: Applied the same changes in `x/bonds/internal/keeper/batch.go`

File: x/bonds/handler.go (hunk 1 above)
Comment (migueldingli1997) [reply x/bonds/handler.go]: Applied the same changes in `x/bonds/internal/keeper/batch.go`

**Pull request merged**